### PR TITLE
[Internal] Preview Build: Add preview build as part of gates

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template
## Description

This adds a build with the preview flag as part of the checks to ensure that preview build is always valid.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Closing issues

Put closes #XXXX in your comment to auto-close the issue that your PR fixes (if such).

## Assignee

Please add yourself as the assignee

## Projects

Please add relevant projects so this issue can be properly tracked.

### CODE REVIEW

File: templates/build-preview.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    clean: true  # if true, execute `execute git clean -ffdx && git reset --hard HEAD` before fetching
+
+  - task: DotNetCoreCLI@2
+    displayName: Build Microsoft.Azure.Cosmos PREVIEW
+    inputs: 
+      command: build  
+      configuration: $(parameters.BuildConfiguration)
+      nugetConfigPath: NuGet.config
+      projects: Microsoft.Azure.Cosmos.sln 
+      arguments: -p:Optimize=true -p:DefineConstants="DOCDBCLIENT COSMOSCLIENT NETSTANDARD20 PREVIEW"
```
Comment (ealsur) [inline templates/build-preview.yml:27]: nit: I think you can just do 
```suggestion
      arguments: -p:Optimize=true -p:IsPreview=true
```

Comment (ealsur) [reply templates/build-preview.yml:27]: That runs `dotnet build -p:IsPreview=true` and passes the `IsPreview` flag that sets the required constants.

**Comment by kirankumarkolli** [comment]
Preview/Internal contract gates will complete the gates contract requirements

**Comment by kirankumarkolli** [comment]
Thanks @j82w  for this.

**Pull request merged**